build.assets/shacalc.py

```diff
@@ -0,0 +1,79 @@
+'''
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+
+"""
+Calculates a hash for the given subdirectories by hashing the hashes of
+each file together.
+"""
```

Comment:
This should be (part of) the docstring for the module; the license blurb could be just comments.